### PR TITLE
Add UI test cases #196

### PR DESCRIPTION
**Pull request by michelleXue**
Adding UI test cases with 60% code coverage.
issue#196

### CODE REVIEW

**Comment by sardorocket** [comment]
will you merge this? @jleandroperez

**Comment by alaskaa30** [comment]
Ok


Envoyé depuis mon téléphone intelligent Samsung Galaxy.
-------- Message d'origine --------De : Sardorbek <notifications@github.com> Date : 18-05-04  10:14 a.m.  (GMT-05:00) À : Automattic/simplenote-ios <simplenote-ios@noreply.github.com> Cc : Subscribed <subscribed@noreply.github.com> Objet : Re: [Automattic/simplenote-ios] Add UI test cases #196 (#197) 
will you merge this? @jleandroperez

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Automattic/simplenote-ios","title":"Automattic/simplenote-ios","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Automattic/simplenote-ios"}},"updates":{"snippets":[{"icon":"PERSON","message":"@sardorocket in #197: will you merge this? @jleandroperez "}],"action":{"name":"View Pull Request","url":"https://github.com/Automattic/simplenote-ios/pull/197#issuecomment-386614900"}}}

**Comment by roundhill** [comment]
Sorry I missed this @michelleXue. I love the idea of adding UI tests! The tests seem to all fail for me, though. It doesn't appear to be successfully logging in to the test account.

Also, I think you didn't commit your changes to the project file. The test folder and files are there but no references to them are in the project.

File: SimplenoteUITests/SimplenoteUITests.swift [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+//  Created by Xue Qin on 1/31/18.
+//  Copyright © 2018 Automattic. All rights reserved.
+//
+
+import XCTest
+
+class SimplenoteUITests: XCTestCase {
+    
+    let app = XCUIApplication()
+    let username = "ui.privacy.utsa4@gmail.com"
```
Comment (roundhill) [inline SimplenoteUITests/SimplenoteUITests.swift]: Was this account created using the `history-analyst-dad` app id? It's a little weird if we have a username/password in the source code for security reasons.

File: SimplenoteUITests/SimplenoteUITests.swift [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+        super.setUp()
+        
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+        
+        // In UI tests it is usually best to stop immediately when a failure occurs.
+        continueAfterFailure = false
+        // UI tests must launch the application that they test. Doing this in setup will make sure it happens for each test method.
+        app.launch()
+        
+        if app.staticTexts["Welcome to Simplenote"].exists {
```
Comment (roundhill) [inline SimplenoteUITests/SimplenoteUITests.swift]: I think we need to completely reset the app here, as the welcome screen doesn't show again if you sign out. It only shows one time ever to the user.

File: SimplenoteUITests/SimplenoteUITests.swift [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+        while element.exists == false
+        {
+            app.swipeUp()
+        }
+        
+        sleep(3)
+    }
+    
+    func logout() {
+
```
Comment (roundhill) [inline SimplenoteUITests/SimplenoteUITests.swift]: Extra whitespace here

File: SimplenoteUITests/SimplenoteUITests.swift [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+        spnoteeditorviewNavigationBar.buttons["Menu"].tap()
+        app.buttons["Markdown toggle"].tap()
+        app.buttons["Pin toggle"].tap()
+        app.buttons["Done"].tap()
+        spnoteeditorviewNavigationBar.buttons["Notes"].tap()
+        
+ 
+    }
+    
+    /*
```
Comment (roundhill) [inline SimplenoteUITests/SimplenoteUITests.swift]: You can remove all of this commented code.

**Comment by roundhill** [comment]
Closing due to inactivity.

**Pull request closed without merging**